### DISCUSSION TITLE
An incomplete cp genome assembly.

### DISCUSSION THREAD

**Discussion by SYSU-SinLee** [Kinggerm/GetOrganelle#161] [Q&A]
Hi, I'm trying to assemble the chloroplast genome of one dicotyledon plant. 

First, I ran GetOrganelle with the parameters `-F embplant_pt  -R 10 -t 1 -k 21,45,65,85,105`, and got an incomplete assembly (the log file, gfa and fastg files are attached with the prefix "run1").

Then I increased the number of rounds to run and added a reference cp genome from a close related species with parameters `-F embplant_pt -o plastome -R 20 -t 10 -k 21,45,65,85,105 -s ../Ref.fa`. The results look better but are still incomplete (files with the prefix "run2").

To improve the completeness of the assembly, I increased the round to run again and also increased the disentangle time limit with parameters `-F embplant_pt -o plastome -R 40 -t 10 -k 21,45,65,85,105 -s ../Ref.fa --disentangle-time-limit 72000`. But the assembly is not circular (files with the prefix "run3" and the figures shown below).

Now I'm trying to run GetOrganelle with an increased value of "-w". And I'll appreciate it if there are any suggestions for this problem. Thanks.

Best regards,
Sin Lee

**Comment by SYSU-SinLee** [comment]
To make it clear and in case it is inconvenient to download and decompress, here are the figures and log files.

**The first run:**
`-F embplant_pt -R 10 -t 1 -k 21,45,65,85,105`
![run1_fastg](https://user-images.githubusercontent.com/26915154/162866508-4f9f6135-76cb-4ff5-ba36-e9418f63c162.png)
![run1_gfa](https://user-images.githubusercontent.com/26915154/162866555-12a78de5-30be-4110-be1b-9341f340fa4c.png)
[run1.log.txt](https://github.com/Kinggerm/GetOrganelle/files/8469013/run1.log.txt)

**The second run:**
`-F embplant_pt -o plastome -R 20 -t 10 -k 21,45,65,85,105 -s ../Ref.fa`
![run2_gfa](https://user-images.githubusercontent.com/26915154/162866624-72de0798-c54a-4afd-9e19-f1737be5008a.png)
![run2_fastg](https://user-images.githubusercontent.com/26915154/162866629-26865fd8-b765-4ef2-8245-a44eb3068cd3.png)
[run2.log.txt](https://github.com/Kinggerm/GetOrganelle/files/8469014/run2.log.txt)

**The third run:**
`-F embplant_pt -o plastome -R 40 -t 10 -k 21,45,65,85,105 -s ../Ref.fa --disentangle-time-limit 72000`
![run3_fastg](https://user-images.githubusercontent.com/26915154/162866666-30cc395b-0283-45b1-9a12-a9b853be0bd1.png)
[run3.log.txt](https://github.com/Kinggerm/GetOrganelle/files/8469015/run3.log.txt)
![run3](https://user-images.githubusercontent.com/26915154/162866678-cee5f70f-6235-4836-8a05-ae6b15441642.png)

Thanks for your time.

**Comment by SYSU-SinLee** [comment]
Here are the results with `-w 95`.
`-F embplant_pt -o plastome -R 40 -t 10 -k 21,45,65,85,105 -s ../Ref.fa --disentangle-time-limit 72000 -w 95`

GFA graph:
![run7_gfa](https://user-images.githubusercontent.com/26915154/162867842-71d36564-7724-4b1a-919a-9a8d6f3f15b2.png)

Fastg graph:
![run7_fastg](https://user-images.githubusercontent.com/26915154/162867860-50799e4c-d9ee-4805-b284-7284a0489997.png)

Log:
[run7.log.txt](https://github.com/Kinggerm/GetOrganelle/files/8469045/run7.log.txt)

**Comment by Kinggerm** [comment]
Please follow the guidelines here: https://github.com/Kinggerm/GetOrganelle/wiki/FAQ#what-should-i-do-with-incomplete-resultbroken-assembly-graph. The top-recommended options are not tried or misunderstood. 

1. The top option was reducing the word size, but you were increasing the word size from the automatically-estimated 73 to 95, which worsened the result.
2. You have 20Gb reads, but all the runs were using 2.37 Gb reads, see `--reduce-reads-for-coverage` or `--max-reads` in  https://github.com/Kinggerm/GetOrganelle/wiki/Usage#main-scripts or through `get_organelle_from_reads.py --help`.


Next time, please turn on the depths and lengths of contigs in Bandage. Also, I recommend only uploading the *.png rather than the *.gfa or *.fastg files to protect your data privacy.

**Reply by SYSU-SinLee** [reply]
Thanks for your time. I've followed your suggestions and tried to adjust the parameters as follows.
`-R 40 -t 5 -k 21,45,65,85 -s ../REF.fa --disentangle-time-limit 72000 -w 60 --max-reads inf --reduce-reads-for-coverage inf --max-extending-len 6000`
 Now it seems the assembly is much better. 
[run11.log.txt](https://github.com/Kinggerm/GetOrganelle/files/8588237/run11.log.txt)
![graph](https://user-images.githubusercontent.com/26915154/165887735-76f3c7de-26e7-4601-947f-37f0383ecdae.png)

**Reply by Kinggerm** [reply]
Congrats! Now you have the complete circular genome.

BTW, `--max-reads inf --reduce-reads-for-coverage inf` is plugging too much data into GetOrganelle and drastically wasting computational time. The time can be much improved to be shorter by picking a number between `the default` and `the infinite`.

**Comment by SYSU-SinLee** [comment]
Hi, I’m trying to assemble the chloroplast genome of a close related species with similar parameters except `--reduce-reads-for-coverage 500` as suggested by the manual. Then I found only a very small fraction of the reads were used and the assembly was highly fragmented. Then I tried a series of coverage settings, but the assemblies are all nearly circular with a breakpoint, even when I used all the reads. I have no idea if I should continue to reduce the word size or increase rounds because it will cost such a long time. And I don't know if the reference genome really helps because it's not mentioned where it was used by GetOrganelle in the log file. I‘ll appreciate it if you have any suggestions. Thanks for your time.

For example, when 17.2 million reads were used. (There graph of .fastg is too complicated to show names and lengths on it.)
![run9 fastg](https://user-images.githubusercontent.com/26915154/166862346-7fc6df43-449a-4e1b-bc8b-6bc5e9a41853.png)
![run9 gfa](https://user-images.githubusercontent.com/26915154/166862378-ca4acf2f-71a3-40bf-b494-6929980e58d1.png)
[run9.get_org.log.txt](https://github.com/Kinggerm/GetOrganelle/files/8628642/run9.get_org.log.txt)

When all the 380 million reads were used.
![run2 gfa](https://user-images.githubusercontent.com/26915154/166862401-c0bc4f0d-aa51-41fc-9980-da769756d0f0.png)
![run2 fastg](https://user-images.githubusercontent.com/26915154/166862405-d5d241b8-b4b0-4a8e-a930-98d251c5ec7c.png)
[run2.log.txt](https://github.com/Kinggerm/GetOrganelle/files/8628646/run2.log.txt)

**Reply by Kinggerm** [reply]
These two graphs are NOT organelle-sufficient. I recommend finetuning the parameters further following https://github.com/Kinggerm/GetOrganelle/wiki/FAQ#what-should-i-do-with-incomplete-resultbroken-assembly-graph
